### PR TITLE
codeintel: Clean up service initialization

### PR DESCRIPTION
**Pull request by efritz**
Small refactor to make each code intel service initializer consistent.

## Test plan

N/A.

### CODE REVIEW

File: internal/codeintel/autoindexing/init.go [hunk 1]
```diff
@@ -3,59 +3,45 @@ package autoindexing
...
-
-		svc = newService(s, uploadSvc, gitserver, sclient, repoUpdater, inf, oc("service"))
+		svc = newService(
+			store.New(db, scopedContext("store")),
+			uploadSvc,
+			gitserver,
+			symbols.DefaultClient,
+			repoUpdater,
+			inference.GetService(db),
+			scopedContext("service"),
```
Comment (Numbers88s) [inline internal/codeintel/autoindexing/init.go:38]:
```suggestion
			observation.ScopedContext("codeintel", "autoindexing", "service"),
```

might as well, you're inlining everything else

Comment (efritz) [reply internal/codeintel/autoindexing/init.go:38]: The other services have more repetition that makes this make sense.

File: internal/codeintel/uploads/init.go [hunk 2]
```diff
@@ -3,47 +3,39 @@ package uploads
...
+func GetService(
+	db, codeIntelDB database.DB,
+	gsc GitserverClient,
+) *Service {
 	svcOnce.Do(func() {
-		logger := log.Scoped(
-			"uploads",
-			"codeintel uploads service",
+		svc = newService(
+			store.New(db, scopedContext("store")),
```
Comment (Numbers88s) [inline internal/codeintel/uploads/init.go:27]: I know this is technically correct, but I prefer how it was before by initializing each one then passing the variables. IMO is cleaner and it seems simpler. Don't expect you to change it. I just wanted to express my preferences.

Comment (efritz) [reply internal/codeintel/uploads/init.go:27]: I can revert them all to locals first, I just wanted to make them consistent. I can do that pre-merge if everything else looks good.

File: internal/observation/observation.go [hunk 3]
```diff
@@ -40,6 +41,23 @@ type Context struct {
 // TestContext is a behaviorless Context usable for unit tests.
 var TestContext = Context{Logger: log.NoOp(), Registerer: metrics.TestRegisterer}
 
+// ContextWithLogger creates a live Context with the given logger instance.
+func ContextWithLogger(logger log.Logger) *Context {
```
Comment (Numbers88s) [inline internal/observation/observation.go:45]: Nice this is helpful. Get out of the way boilerplate 😄

**Review by Numbers88s — APPROVED** [review]

File: internal/codeintel/uploads/init.go (hunk 2 above)
Comment (efritz) [reply internal/codeintel/uploads/init.go]: [9946294](https://github.com/sourcegraph/sourcegraph/pull/42507/commits/99462941312526ca78858737f361f6b8b3c7de68)

Comment (Numbers88s) [reply internal/codeintel/uploads/init.go]: Dope man, thanks for doing that 🚀 🙏

**Pull request merged**